platform/pensando/sonic-platform-modules-dpu/dpu/utils/dpu_pensando_util.py

```diff
@@ -0,0 +1,172 @@
+
+def set_ubootenv_config():
+    cmd = "cat /proc/mtd | grep -e 'ubootenv' | awk '{print $1}' | tr -dc '0-9'"
+    mtd_ubootenv = run_cmd(cmd)
+    fw_env_config = "/dev/mtd{} 0x0 0x1000 0x10000".format(mtd_ubootenv)
+    with open("/etc/fw_env.config","w") as f:
+        f.write(fw_env_config)
+
+def start_pmon():
+    cmd = "systemctl is-enabled pmon.timer"
+    output = run_cmd(cmd)
+    if output == "disabled":
+        try:
+            cmd = "systemctl enable pmon.timer"
+            run_cmd(cmd)
+        except:
+            pass
+    elif output == "static":
+        try:
+            cmd = "systemctl enable pmon.timer"
+            run_cmd(cmd)
+        except:
+            pass
+    else:
+        pass
+    cmd = "systemctl is-enabled pmon"
+    output = run_cmd(cmd)
+    if output == "static":
+        try:
+            cmd = "systemctl start pmon"
+            run_cmd(cmd)
+        except:
+            pass
+
+def configure_iptable_rules():
+    try:
+        iptable_cfg_cmd = "sudo iptables-legacy -D tcp_inbound -p tcp --dport 11357:11360 -j DROP"
+        run_cmd(iptable_cfg_cmd)
+    except:
+        pass
+
+def pcie_tx_setup():
+    dpu_slot_id = int(run_cmd("cpldapp -r 0xA").strip(), 16)
+    if dpu_slot_id == 6 or dpu_slot_id == 7:
+        run_cmd("docker exec polaris /nic/tools/run-aacs-server.sh -p 9001 export SERDES_DUT_IP=localhost:9001")
+        run_cmd("docker exec -e SERDES_ADDR=1:1-1:3f -e SERDES_DUT_IP=localhost:9001 -e SERDES_SBUS_RINGS=4 polaris aapl serdes -addr 1:39 -pre 0 -post 0 -atten 10")
+        run_cmd("docker exec -e SERDES_ADDR=1:1-1:3f -e SERDES_DUT_IP=localhost:9001 -e SERDES_SBUS_RINGS=4 polaris aapl serdes -addr 1:3b -pre 0 -post 0 -atten 10")
+        run_cmd("docker exec -e SERDES_ADDR=1:1-1:3f -e SERDES_DUT_IP=localhost:9001 -e SERDES_SBUS_RINGS=4 polaris aapl serdes -addr 1:3d -pre 0 -post 0 -atten 10")
+        run_cmd("docker exec -e SERDES_ADDR=1:1-1:3f -e SERDES_DUT_IP=localhost:9001 -e SERDES_SBUS_RINGS=4 polaris aapl serdes -addr 1:3f -pre 0 -post 0 -atten 10")
+        run_cmd("docker exec -e SERDES_ADDR=1:1-1:3f -e SERDES_DUT_IP=localhost:9001 -e SERDES_SBUS_RINGS=4 polaris aapl dev -addr 1:39 -v 1")
+        run_cmd("docker exec -e SERDES_ADDR=1:1-1:3f -e SERDES_DUT_IP=localhost:9001 -e SERDES_SBUS_RINGS=4 polaris aapl dev -addr 1:3b -v 1")
+        run_cmd("docker exec -e SERDES_ADDR=1:1-1:3f -e SERDES_DUT_IP=localhost:9001 -e SERDES_SBUS_RINGS=4 polaris aapl dev -addr 1:3d -v 1")
+        run_cmd("docker exec -e SERDES_ADDR=1:1-1:3f -e SERDES_DUT_IP=localhost:9001 -e SERDES_SBUS_RINGS=4 polaris aapl dev -addr 1:3f -v 1")
+
+def main():
+    #All init routines to be written below
+    cmd = 'PATH=$PATH:/usr/sbin/'
+    run_cmd(cmd)
+    try:
+        set_ubootenv_config()
+    except:
+        pass
+    time.sleep(10)
+    configure_iptable_rules()
+    fetch_dpu_files()
+    time.sleep(5)
+    set_onie_version()
+    cp_to_shared_mem()
+    set_cpldapp()
+    start_pmon()
```

Comment:
@shanshri PMON should be orchestrated by systemd right? Why a different approach?

Reply:
@prgeor, can you please point me to correct approach? Right now without it, pmon container is not coming up. We have to start it manually.